File: PYTHON/ArrayAndStrings/Easy/TwoSum.py
def TwoSum(nums,target):
    num_to_index = {}
    for index, num in enumerate(nums):
        compliment = target - num
        if compliment in num_to_index:
            return [num_to_index[compliment], index]
        num_to_index[num] = index
    return None



# Example usage:    
nums = [2, 7, 11, 15]
target = 9
result = TwoSum(nums,target)
if result:
    print(f"Indices of the two numbers that add up to {target} are: {result}")
else:
    print("No two numbers add up to the target.")

File: JAVASCRIPT/ArrayAndStrings/Easy/TwoSum.js
// TWO SUM
function TwoSum(nums,target){
    let map = new Map();
    for(let i=0;i<nums.length;i++){
        let compliment = target - nums[i];
        if(map.has(compliment)){
            return [map.get(compliment),i];
        }
        map.set(nums[i], i)
    }
}

// Example usage:
let nums = [2, 7, 11, 15];
let target = 9;
let result = TwoSum(nums, target);
console.log(result); // Output: [0, 1]